cns/service/main.go

```diff
@@ -1252,3 +1253,108 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
+		logger.Printf("Error while executing pwsh command to get VLAN Tag advance property of %s: %v", adapatername, err)
+		return
+	}
+	if adapterNamewithVLANTag != "" {
+		logger.Printf("Found %s in adapter's advanced properties", priorityVLANTagIdentifier)
+
+		//set reg value for Priorityvlantag of adapater to 3
+		cmd = fmt.Sprintf("Set-NetAdapterAdvancedProperty -Name \"%s\" -RegistryKeyword \"%s\" -RegistryValue 3", adapatername, priorityVLANTagIdentifier)
+		_, err := platform.ExecutePowershellCommand(cmd)
+		if err != nil {
+			logger.Printf("Error while setting up registery value for PriorityVLANTag for adapater: %v", err)
+		}
+		//check if value is set or not
+		cmd = fmt.Sprintf(" Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty RegistryValue", priorityVLANTagIdentifier, adapatername)
+		regvalue, err := platform.ExecutePowershellCommand(cmd)
+		if err != nil {
+			logger.Printf("Error while checking registery value for PriorityVLANTag for adapater: %v", err)
+		}
+		if regvalue != "3" {
+			logger.Printf("Failure while setting Mellanox Network Adapter: \"%s\" with \"%s\" property value as 3", adapatername, priorityVLANTagIdentifier)
+		} else {
+			logger.Printf("Successfully set Mellanox Network Adapter: \"%s\" with \"%s\" property value as 3", adapatername, priorityVLANTagIdentifier)
+		}
+	} else {
+		//looks like its version 3 so vlantag not available so need to follow other route
+		logger.Printf("Could not find \"%s\" in adapter's advanced properties", priorityVLANTagIdentifier)
+		logger.Printf("Proceeding in a different way")
+		logger.Printf("Searching through CIM instances for Network devices with \"%s\" in the name", mellanoxSearchString)
+		cmd = fmt.Sprintf("Get-CimInstance -Namespace root/cimv2 -ClassName Win32_PNPEntity | Where-Object PNPClass -EQ \"Net\" | Where-Object { $_.Name -like \"%s\" } | Select-Object -ExpandProperty DeviceID", mellanoxSearchString)
+		deviceid, err := platform.ExecutePowershellCommand(cmd)
+		if err != nil {
+			logger.Printf("Error while executing pwsh command to get device id of %s: %v", adapatername, err)
+			return
+		}
+		if deviceid != "" {
+			logger.Printf("Device ID found: %s", deviceid)
+			logger.Printf("Getting Pnp Device properites for %s", deviceid)
+			cmd = fmt.Sprintf("Get-PnpDeviceProperty -InstanceId \"%s\" | Where-Object KeyName -EQ \"DEVPKEY_Device_Driver\" | Select-Object -ExpandProperty Data", deviceid)
+			registryKeySuffix, err := platform.ExecutePowershellCommand(cmd)
+			if err != nil {
+				logger.Printf("Error while executing pwsh command to get device id of %s: %v", adapatername, err)
+				return
+			}
+			logger.Printf("Registry key suffix found: %s", registryKeySuffix)
+			registryKeyFullPath := registryKeyPrefix + registryKeySuffix
+			logger.Printf("Registry key full path: %s", registryKeyFullPath)
+			logger.Printf("Updating %s to be 3", priorityVLANTagIdentifier)
```

Comment:
no need to log this upfront. It can be logged as error if unable to update to 3

Reply:
sure